lib/bundler/fetcher/compact_gem_list/cache.rb

```diff
@@ -0,0 +1,79 @@
+module Bundler
+  class Fetcher::CompactGemList
+    class Cache
+      attr_reader :directory
+
+      def initialize(directory)
+        @directory = p Pathname(directory).expand_path
+        FileUtils.mkdir_p dependencies_path(nil)
+      end
+
+      def names
+        lines(names_path)
+      end
+
+      def names_path
+        directory + 'names'
+      end
+
+      def versions
+        versions_by_name = {}
+        lines(versions_path).map do |line|
+          name, versions_string = line.split(" ", 2)
+          versions_string.split(",").map! do |version|
+            version.split("-", 2).unshift(name)
+          end
+        end
+        versions_by_name
+      end
+
+      def versions_path
+        directory + 'versions'
+      end
+
+      def dependencies(name)
+        lines(dependencies_path(name)).map do |line|
+          parse_gem(line)
+        end
+      end
+
+      def dependencies_path(name)
+        directory + 'dependencies' + name.to_s
+      end
+
+      def specific_dependency(name, version, platform)
+        pattern = [version, platform].compact!.join("-")
+        matcher = %r{\A#{Regexp.escape(pattern)} } unless pattern.empty?
+        lines(dependencies_path(name)).each do |line|
+          return parse_gem(line) if line =~ matcher
+        end if matcher
+        nil
+      end
+
+      private
+
+      def lines(path)
+        return [] unless path.file?
+        lines = path.read.lines
+        header = lines.index("---\n")
+        lines = header ? lines[header+1..-1] : lines
+        lines.map!(&:strip!)
```

Comment:
yeah, strip! will return nil if there's nothing to strip, and that's dangerous here :P